src/main/scala/com/gu/i18n/CountryGroup.scala

```diff
@@ -312,4 +312,19 @@ object CountryGroup {
 
   def availableCurrency(currencies: Set[Currency])(country: Country) =
     byCountryCode(country.alpha2).map(_.currency).filter(currencies)
+
+  def byOptimisticCountryNameOrCode(str: String): Option[Country] = {
+    val clean = str.replace(".", "")
+    val asCode = clean.toUpperCase
+    val name = clean.toLowerCase
+
+    countryByName(name) orElse countryByCode(asCode) orElse (name match {
+      case _ if name endsWith "ireland" => Some(Country.Ireland)
+      case _ if asCode == "USA" => Some(Country.US)
+      case _ if name startsWith "the" => countryByName(name.replaceFirst("the", "").trim)
+      case _ if name == "russian federation" => countryByName("Russia")
+      case _ => None
```

Comment:
Let's add replacements for these countries too (then we can deprecate the mappings in membership-workflow:DataProcessing:

"gb"
"GREAT BRITAIN"
"VIET NAM"

Reply:
If we want this to replace the [findCountry method](https://github.com/guardian/membership-workflow/blob/master/app/model/DataProcessing.scala#L44) in membership workflow it should also support 3 letter iso country codes.